pytest_rts/utils/mappinghelper.py

```diff
@@ -0,0 +1,302 @@
+"""Module for MappingHelper class"""
+import os
+from typing import Dict, NamedTuple, Set, Tuple
+from coverage import CoverageData
+from _pytest.python import Function
+
+TestrunData = NamedTuple(
+    "TestrunData",
+    [
+        ("pytest_item", Function),
+        ("elapsed_time", float),
+        ("coverage_data", CoverageData),
+        ("found_testfiles", Set[str]),
+        ("test_function_lines", Dict[str, Dict[str, Tuple[int, int]]]),
+    ],
+)
+
+TestFunctionData = NamedTuple(
+    "TestFunctionData",
+    [
+        ("testfile_id", int),
+        ("testname", str),
+        ("start_line_number", int),
+        ("end_line_number", int),
+        ("elapsed_time", float),
+    ],
+)
+
+
+class MappingHelper:
+    """Class to handle mapping database initialization and updating"""
+
+    def __init__(self, connection):
+        """Set the connection"""
+        self.connection = connection
+
+    @property
+    def srcfiles(self):
+        """Tuple of id and path to a source code file in the mapping database"""
+        return [
+            (x[0], x[1])
+            for x in self.connection.execute("SELECT id, path FROM src_file").fetchall()
+        ]
+
+    @property
+    def testfiles(self):
+        """Tuple of id and path to a test code file in the mapping database"""
+        return [
+            (x[0], x[1])
+            for x in self.connection.execute(
+                "SELECT id, path FROM test_file"
+            ).fetchall()
+        ]
+
+    @property
+    def last_update_hash(self):
+        """Git commit hash of latest database update"""
+        return self.connection.execute("SELECT hash FROM last_update_hash").fetchone()[
+            0
+        ]
+
+    def __delete_lines(self, line_ids, file_id):
+        """Remove entries from covered line mapping"""
+        entries = [(line_id, file_id) for line_id in line_ids]
+        self.connection.executemany(
+            "DELETE FROM test_map WHERE line_id == ? AND file_id == ?",
+            entries,
+        )
+
+    def __add_lines(self, src_id, test_function_id, lines):
+        """Add entries to covered line mapping"""
+        entries = [(src_id, test_function_id, line) for line in lines]
+        self.connection.executemany(
+            "INSERT OR IGNORE INTO test_map (file_id,test_function_id,line_id) VALUES (?,?,?)",
```

Comment:
We really, really should stop ~~doing sunset manually~~ writing SQL as string constants, and start using a proper tool (e.g. sqlalchemy). Not for this PR, though.